### PR TITLE
fix module that it reports the correct status of a change

### LINKED ISSUE #64371: postgresql_privs always reports as changed when using default_privs
##### SUMMARY
The `postgresql_privs` module always reports as changed if the `type` option is `default_privs`. Ansible adds multiple psql commands before running the actual task (see the verbatim command output bellow), I don't know why it's implemented like this, but such behavior leads to unexpected results: My issue isn't that risky, but if you try to revoke a privilege from a `sequence` (using `state: absent`), it will also revoke all default `table` and `type` privileges.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
postgresql_privs

##### ANSIBLE VERSION
```
ansible 2.8.6
  config file = /home/razor/ansible-bug/ansible.cfg
  configured module search path = ['/home/razor/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/razor/.pyenv/versions/3.7.0/lib/python3.7/site-packages/ansible
  executable location = /home/razor/.pyenv/versions/3.7.0/bin/ansible
  python version = 3.7.0 (default, Nov  3 2019, 02:20:02) [GCC 4.8.4]
```

##### CONFIGURATION
```
DEFAULT_HOST_LIST(/home/razor/ansible-bug/ansible.cfg) = ['/home/razor/ansible-bug/inventories/development']
```

##### OS / ENVIRONMENT
Ubuntu 18.04

##### STEPS TO REPRODUCE
Create a test.yml file with the playbook below:

```yaml
- name: Configure and deploy PostgreSQL
  hosts: all
  become: yes
  vars:
    - role_name: role_test
    - database_name: database_test
    - postgresql_privs:
      - roles: "{{ role_name }}"
        db: "{{ database_name }}"
        privileges:
          - privs: SELECT,INSERT,UPDATE,DELETE
            type: default_privs
            objs: TABLES
            schema: public
          - privs: SELECT,UPDATE
            type: default_privs
            objs: SEQUENCES
            schema: public
  tasks:
    - name: PostgreSQL databases
      postgresql_db:
        name: "{{ database_name }}"
      become_user: postgres

    - name: PostgreSQL users
      postgresql_user:
        name: "{{ role_name }}"
      become_user: postgres

    - name: PostgreSQL privileges
      postgresql_privs:
        roles: "{{ item.0.roles }}"
        db: "{{ item.0.db }}"
        privs: "{{ item.1.privs }}"
        type: "{{ item.1.type }}"
        objs: "{{ item.1.objs | default(omit) }}"
        schema: "{{ item.1.schema | default(omit) }}"
        state: "{{ item.1.state | default(item.0.state | default('present')) }}"
      with_subelements:
        - "{{ postgresql_privs }}"
        - privileges
      become_user: postgres
```

Run the following command **twice**:
```
ansible-playbook test.yml
```

##### EXPECTED RESULTS
The task "PostgreSQL privileges" should report as "ok" at the second run.

##### ACTUAL RESULTS
The task "PostgreSQL privileges" always shows "changed".

```paste below
TASK [Ensure PostgreSQL privileges are present] ***************************************************************************************************************
changed: [acme.test] => (item=[{'roles': 'role_test', 'db': 'database_test'}, {'privs': 'SELECT,INSERT,UPDATE,DELETE', 'type': 'default_privs', 'objs': 'TABLES', 'schema': 'public'}]) => {"ansible_loop_var": "item", "changed": true, "item": [{"db": "database_test", "roles": "role_test"}, {"objs": "TABLES", "privs": "SELECT,INSERT,UPDATE,DELETE", "schema": "public", "type": "default_privs"}], "queries": ["ALTER DEFAULT PRIVILEGES IN SCHEMA public REVOKE ALL ON TABLES FROM \"role_test\";\nALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT UPDATE,INSERT,SELECT,DELETE ON TABLES TO \"role_test\";\nALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT USAGE ON TYPES TO \"role_test\";"]}
changed: [acme.test] => (item=[{'roles': 'role_test', 'db': 'database_test'}, {'privs': 'SELECT,UPDATE', 'type': 'default_privs', 'objs': 'SEQUENCES', 'schema': 'public'}]) => {"ansible_loop_var": "item", "changed": true, "item": [{"db": "database_test", "roles": "role_test"}, {"objs": "SEQUENCES", "privs": "SELECT,UPDATE", "schema": "public", "type": "default_privs"}], "queries": ["ALTER DEFAULT PRIVILEGES IN SCHEMA public REVOKE ALL ON SEQUENCES FROM \"role_test\";\nALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT UPDATE,SELECT ON SEQUENCES TO \"role_test\";\nALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT USAGE ON TYPES TO \"role_test\";"]}
```

### PR DESCRIPTION
**Pull request by tcraxs**
(cherry picked from commit 2028734ac334323aff4b9ac7289857e5f74bef53 and 426e37ea92db037fe9367a6daa4d17622b1faf1d)

Backport of https://github.com/ansible/ansible/pull/65061 and https://github.com/ansible/ansible/pull/65125

##### SUMMARY
`postgresql_privs` now reports the correct status of a change when using `type=default_privs`

"Fixes #64371 "

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
postgresql_privs

### CODE REVIEW

**Comment by samdoran** [comment]
Hold off on merging this until we get #65125 merged in `devel`.

**Comment by samdoran** [comment]
We need to add 426e37ea92 to this PR.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Could you please include change https://github.com/ansible/ansible/commit/426e37ea92db037fe9367a6daa4d17622b1faf1d ? Thanks.

**Comment by tcraxs** [comment]
@samdoran thanks for the info, I will include it :)

**Review by Akasurde — APPROVED** [review]
LGTM

**Review by Andersson007 — APPROVED** [review]
good job!
shipit

**Comment by Andersson007** [comment]
cc @nitzmahone @mattclay

**Comment by tcraxs** [comment]
@samdoran @Akasurde can someone merge it please?

**Comment by Akasurde** [comment]
> @samdoran @Akasurde can someone merge it please?

Since this is a backport PR, it will merged by release manager.

**Comment by mattclay** [comment]
Why are there two backports this PR? It's much easier for RMs to review backports that have a 1:1 relationship with the original devel PR.

**Comment by tsacha** [comment]
Hey, I had this error on Ansible 2.9.2 (it works on 2.9.1). Is this related?

`
{"attempts": 3, "changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 102, in <module>\n  File \"<stdin>\", line 94, in _ansiballz_main\n  File \"<stdin>\", line 40, in invoke_module\n  File \"/usr/lib/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_postgresql_privs_payload_gsems1r4/ansible_postgresql_privs_payload.zip/ansible/modules/database/postgresql/postgresql_privs.py\", line 1057, in <module>\n  File \"/tmp/ansible_postgresql_privs_payload_gsems1r4/ansible_postgresql_privs_payload.zip/ansible/modules/database/postgresql/postgresql_privs.py\", line 1038, in main\n  File \"/tmp/ansible_postgresql_privs_payload_gsems1r4/ansible_postgresql_privs_payload.zip/ansible/modules/database/postgresql/postgresql_privs.py\", line 752, in manipulate_privs\nTypeError: '<' not supported between instances of 'NoneType' and 'NoneType'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
`

Task:

```
    - name: Create permissions for {{ client }}
      postgresql_privs:
        db: '{{ client }}'
        roles: '{{ client }}'
        privs: ALL
        objs: ALL_IN_SCHEMA
      register: result
      retries: 3
      delay: 10
      until: result is not failed
      become: true
      become_user: postgres
```

**Comment by Andersson007** [comment]
@tsacha hey, probably yes and we need conditions there with .sort()
```
        executed_queries.append(query)
        self.cursor.execute(query)
        status_after = get_status(objs)
        status_before.sort()
        status_after.sort()
        return status_before != status_after
```
will see

**Comment by Andersson007** [comment]
@tsacha thanks for reporting this!
I created an issue https://github.com/ansible/ansible/issues/65761
If you can add anything else important, please, do it there, maybe any other ditailes which could be important to clerify the whole situation, see the issue's sections in desctiption as a hint.

**Comment by Andersson007** [comment]
https://github.com/ansible/ansible/pull/65903

**Pull request merged**
